request-server.go

```diff
@@ -24,12 +25,17 @@ type Handlers struct {
 	FileList FileLister
 }
 
+type openRequest struct {
```

Comment:
Why a new type and not just use Request?

Reply:
I don't have a problem adding the cancel function to the Request itself but it seemed a little out of place since the sftp request handler code is keeping track of the open requests. I figured it makes sense to separate tracking of requests from the details about a specific request.

Reply:
Hm… I suppose my “better” idea would be that somehow the cancel function should then be contained through a closure or something… but with the design as is, I am not sure how that would work, so… 🤷‍♀️

Reply:
I'm also not sure about the separate struct just for the request + cancel func, but I get your point. I'll leave it for now so I can see it in context and work with it some.